### PR TITLE
machines: Change Import VM dialog title and buttons (fixes #12614)

### LINKED ISSUE #12614: machines: Dialog to import a VM reads "Create new VM"
Cockpit version: 201
Page: Machines

Dialog to import a VM reads "Create new VM".

image

### PR DESCRIPTION
**Pull request by MetalDent**
This PR fixes #12614

### CODE REVIEW

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx [hunk 1]
```diff
@@ -770,7 +770,7 @@ class CreateVmModal extends React.Component {
             <Modal id='create-vm-dialog' show onHide={ this.props.close }>
                 <Modal.Header>
                     <Modal.CloseButton onClick={ this.props.close } />
-                    <Modal.Title> {`Create New Virtual Machine`} </Modal.Title>
+                    <Modal.Title> {`Import A Virtual Machine`} </Modal.Title>
```
Comment (marusak) [inline pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: This needs the same logic as the change below - that it shows `Create New Virtual Machine` when `mode` is `create` and else it shows `Import A Virtual Machine`.

Also I just noticed that we do not translate this string. When you notice the changes in the the button text, you see that we put the strings into `_()`. Although this seems weird, `_` is a function name. This function translates strings when you use Cockpit in different language than English. Can you please also introduce these changes? Thank you.

Comment (MetalDent) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: Yeah I'll put the similar condition on line 773 but I'm sorry I didn't understand the change regarding the button strings.

**Comment by MetalDent** [comment]
> > This PR fixes #12614
> 
> Please don't close an open PR if you need to make some changes.
> You can force push in the same branch that you used to create the first one.
> In this way we can keep the history of the reviews in one place - and see also in the UI the changes that you did between two review iterations :)

I will keep that in mind :)

**Comment by MetalDent** [comment]
Also I think line [831](https://github.com/cockpit-project/cockpit/blob/master/pkg/machines/components/create-vm-dialog/createVmDialog.jsx#L831) needs to be changed too from "Create VM" to just "Create" and also for "Import VM"
Just a suggestion :)

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx (hunk 1 above)
Comment (marusak) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: >  I'm sorry I didn't understand the change regarding the button strings.

Notice difference between `_("Create")` and `"Create`. The second string is not translated. When you look at those texts in buttons they are translated. When you look at those titles, they are not.

Comment (MetalDent) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: Yeah, I got that. So we don't need to translate as we're writing it only in English, right?

Comment (marusak) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: not really, this is for when someone is using cockpit in different language than English. When the user uses English, then this string is presented as is - but when they pick different language, then we can translate those strings. And we (as developers) mark strings which should be translated into different languages with this construction. And we want to translate also those titles.

**Comment by MetalDent** [comment]
> Also this should be one commit - can you please squash them? There are multiple ways how to do this, maybe a good thing is to learn interactive rebasing.
> `git rebase -i HEAD~3` (the number is how many commits you will work with)
> It opens editor for you and you can edit those selected commits. There is very handy help below those commits :) Just one hint - the third commit (on third line) is the newest one. When you save the changes, then it is applied.

Done!

**Comment by marusak** [comment]
Thanks, I check tests and they found some small issue.
Please check output of `./tools/test-static-code` and adjust to make it happy :)

**Comment by marusak** [comment]
@MetalDent are you stuck on something I can help you with?

**Comment by MetalDent** [comment]
> @MetalDent are you stuck on something I can help you with?

no no, it's not like that. I was having my sem exams. I'll fix this up soon.

Sorry for the delay.

**Comment by MetalDent** [comment]
> Thanks, I check tests and they found some small issue.
> Please check output of `./tools/test-static-code` and adjust to make it happy :)
> 
> Also since it will need one more force push anyway, can you please edit the commit message? (you can do it with `git rebase -i` as well, just use different command. and one more trick, you can use `fixup` instead of `squash` so it does not add the comments :) )
> Anyway, how the message should look like:
> 
> 1. Title should not have ` (fixes #12614)`
> 2. Commit body should not have those message that were added on squashing. It should have at least one line that says only `Fixes #12614`. Also in our team we have convention that we also add on another line `Closes #12807` to keep track of pull requests. Also it is a very good practice to explain in the commit body why these changes, but in this case it is probably not necessary.
> 
> P.S. I highly recommend to read https://chris.beams.io/posts/git-commit/ (to everyone ;) )

All done!

Please check it out.

**Comment by marusak** [comment]
Thank you, 
unfortunately in the meantime there were some big changes and therefore this cannot run tests. Can you please rebase on the current master?

**Comment by marusak** [comment]
Thanks, I was able to let the tests run and they are failing.
It is not that easy to read what is failing at first, but in short:
1. Inspect any failed test (for example `debian-stable`) by clicking on `Details`
2. Failed tests are in red on the top of the file - by clicking on them you can see the logs
3. You should see [this](https://209.132.184.41:8493/logs/pull-12807-20191004-121313-68efdfa0-cockpit-project-cockpit--debian-stable/log.html#69-2) page
4. There are some logs and then there is Traceback, in which you are interested. It says filename, lines and it should be clear what is failing.

Can you please take a look and fix it?

**Comment by marusak** [comment]
Also there is conflict, which needs to be resolved

**Comment by marusak** [comment]
ping @MetalDent

**Pull request merged**